client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/db/constraint/H2ConstraintExtractor.java

```diff
@@ -0,0 +1,132 @@
+package org.evomaster.client.java.controller.internal.db.constraint;
+
+
+import org.evomaster.client.java.controller.api.dto.database.schema.DbSchemaDto;
+import org.evomaster.client.java.controller.api.dto.database.schema.TableDto;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class H2ConstraintExtractor extends TableConstraintExtractor {
+
+    /**
+     * Expects the schema explained in
+     * http://www.h2database.com/html/systemtables.html#information_schema
+     *
+     * @param connectionToH2 a connection to a H2 database
+     * @param schemaDto      a DTO schema with retrieved information from the JBDC metada
+     * @throws Exception
+     */
+    public List<DbTableConstraint> extract(Connection connectionToH2, DbSchemaDto schemaDto) throws SQLException {
+
+        List<DbTableConstraint> columnConstraints = extractColumnConstraints(connectionToH2, schemaDto);
+
+        List<DbTableConstraint> tableCheckExpressions = extractTableConstraints(connectionToH2, schemaDto);
+
+        List<DbTableConstraint> allConstraints = new LinkedList<>();
+        allConstraints.addAll(columnConstraints);
+        allConstraints.addAll(tableCheckExpressions);
+        return allConstraints;
+    }
+
+    /**
+     * For each table in the schema DTO, this method appends
+     * the constraints that are originated in the ALTER TABLE commands
+     * for those particular tables.
+     * <p>
```

Comment:
aren't `CHECK` constraints possible directly on table declaration (ie `CREATE`)  besides `ALTER`?

Reply:
Indeed. The documentation is not clear. I changed the documentation and I also added two more test cases for extracting constraints to clarify the behaviour.